src/math.jl

```diff
@@ -55,6 +55,15 @@ end
 Base.:*(l::AbstractDimensions, r::AbstractDimensions) = map_dimensions(+, l, r)
 Base.:/(l::AbstractDimensions, r::AbstractDimensions) = map_dimensions(-, l, r)
 
+# Multiplying ranges with units
+for (type, _, _) in ABSTRACT_QUANTITY_TYPES
+    @eval begin
+        Base.:*(r::StepRange, q::$type) = (first(r) * q):(Base.step(r) * q):(last(r) * q)
+        Base.:*(r::StepRangeLen, q::$type) = range(first(r) * q; step=Base.step(r) * q, length=length(r))
+        Base.:*(q::$type, r::Union{StepRange,StepRangeLen}) = r * q
```

Comment:
Notice how all ways to do this land at the same `broadcasted` method. That's probably what I'd try to overload, rather than aiming for left & right & broadcasting separately:
```
julia> (1:0.25:4) * (u"inch")
ERROR: Cannot create an additive identity from `Type{<:Quantity}`, as the dimensions are unknown. Please use `zero(::Quantity)` instead.
Stacktrace:
...
  [5] *
    @ ./twiceprecision.jl:308 [inlined]
  [6] broadcasted
    @ ./broadcast.jl:1155 [inlined]
  [7] broadcasted
    @ ./broadcast.jl:1332 [inlined]
  [8] broadcast_preserving_zero_d
    @ ./broadcast.jl:861 [inlined]
  [9] *(A::StepRangeLen{Float64, Base.TwicePrecision{…}, Base.TwicePrecision{…}, Int64}, B::Quantity{Float64, Dimensions{…}})
    @ Base ./arraymath.jl:24

julia> (1:0.25:4) .* (u"inch")
ERROR: Cannot create an additive identity from `Type{<:Quantity}`, as the dimensions are unknown. Please use `zero(::Quantity)` instead.
...
 [5] *
   @ ./twiceprecision.jl:308 [inlined]
 [6] broadcasted
   @ ./broadcast.jl:1155 [inlined]
 [7] broadcasted(::typeof(*), ::StepRangeLen{…}, ::Quantity{…})  <--- this one
   @ Base.Broadcast ./broadcast.jl:1332
 [8] top-level scope
   @ REPL[20]:1
Some type information was truncated. Use `show(err)` to see complete types.

julia> (u"inch") * (1:0.25:4)
ERROR: Cannot create an additive identity from `Type{<:Quantity}`, as the dimensions are unknown. Please use `zero(::Quantity)` instead.
...
  [6] *
    @ ./twiceprecision.jl:316 [inlined]
  [7] broadcasted
    @ ./broadcast.jl:1148 [inlined]
  [8] broadcasted
    @ ./broadcast.jl:1332 [inlined]
  [9] broadcast_preserving_zero_d
    @ ./broadcast.jl:861 [inlined]
 [10] *(A::Quantity{Float64, Dimensions{…}}, B::StepRangeLen{Float64, Base.TwicePrecision{…}, Base.TwicePrecision{…}, Int64})
```

Reply:
> Notice how all ways to do this land at the same broadcasted method. That's probably what I'd try to overload, rather than aiming for left & right & broadcasting separately:

My worry is that this dependence is version dependent, and the syntax may break in the future. Wdyt?

Reply:
You should certainly test the user-facing `λ*r`, `r*λ`, `r.*λ` etc. I think this path to make a new range via broadcasting has been the same forever.

You might also be able to fix it later in the pathway. Maybe you just need to define `Base.TwicePrecision(x::Quantity)` appropriately?

BTW I suspect it should be illegal to have a StepRange of quantities, as OrdinalRange demands:

> The steps should be always-exact multiples of oneunit, and T should be a
  "discrete" type, which cannot have values smaller than oneunit.

Reply:
The `broadcasted` method it's landing on is

```julia
@inline function broadcasted(f::F, arg1, arg2, args...) where {F}
    arg1′ = broadcastable(arg1)
    arg2′ = broadcastable(arg2)
    args′ = map(broadcastable, args)
    broadcasted(combine_styles(arg1′, arg2′, args′...), f, arg1′, arg2′, args′...)
end
```

I feel like it would be hard to maintain if I overload this? (Also not sure how)

Am I looking at the wrong method?

Reply:
Oh yeah TwicePrecision might be much easier. Looking at it now...